### DISCUSSION TITLE
user defined commit prefix

### DISCUSSION THREAD

**Discussion by chrisjohnson00** [fluxcd/flux2#1143] [Proposals] [answered]
Loving Flux2!!
What would make this even easier/better for my organization is if I could define a commit prefix for the "Add flux..." commit messages.  
I've setup EKS and Flux with support for ephemeral clusters, so there's a number of commits incoming for when an ephemeral cluster is spun up. 
Currently I have our CI skipping various prefixes and have added "Add flux" as one of them, but fear that with the evolution of the tooling, this message will change.  
I'm imagining this as part of the bootstrap command execution.
Thoughts?  Insults?

**Comment by stefanprodan** [comment, marked as answer]
@hiddeco could you add a commit prefix flag to your PR? https://github.com/fluxcd/flux2/pull/968

**Reply by hiddeco** [reply]
Prefixing seems odd, I think we either need to give users control over the full commit message, or the commit description (so e.g. `[skip ci]` can be added).

**Reply by chrisjohnson00** [reply]
> Prefixing seems odd, I think we either need to give users control over the full commit message, or the commit description (so e.g. `[skip ci]` can be added).

I'm down with this!

**Reply by hiddeco** [reply]
Added in https://github.com/fluxcd/flux2/pull/968/commits/ee8e5bb43e49dd4d56ea237c130065b7aa498eea

**Reply by chrisjohnson00** [reply]
@hiddeco  - I've started using `--commit-message-appendix` to add `[skip ci]` and that works for not running CI just fine, thanks!
Is there anything planned for being able to customize more of the commit message?

For example I can do this now:
```
Add Flux v0.12.1 component manifests
[skip ci]
```
My company has commit standards enforcement, and these messages don't follow convention.  I'd like to do something like:
```
ci: Add Flux v0.12.1 component manifests
[skip ci]
```

Is there a flag not in the help, or an upcoming enhancement which would allow further customization that would match my 2nd example?

Thanks again, still loving Flux

**Reply by gibbonsjohnm** [reply]
Any update on what @chrisjohnson00 requested or possibly revisiting the idea of a commit prefix? We also have commit standards and need to add a prefix to anything Flux commits to meet them. Thanks in advance!

**Reply by hiddeco** [reply]
Please create a new issue for this request: https://github.com/fluxcd/flux2/issues/new
